src/hooks.js

```diff
@@ -0,0 +1,61 @@
+import * as PIXI from "pixi.js";
+import React from "react";
+import { useContext, useEffect, useRef, useState, useLayoutEffect } from "react";
+import { AppContext } from "./AppProvider";
+import { getCanvasProps } from "./stageProps";
+import { unmount } from "./render";
+
+export function usePixi() {
+  return useContext(AppContext);
+}
+
+export function usePixiTicker(fn) {
+  const app = usePixi();
+
+  useEffect(() => {
+    app.ticker.add(fn);
+
+    return () => {
+      app.ticker.remove(fn);
+    };
+  }, []);
+}
+
+export function usePreviousProps(value) {
+  const ref = useRef({});
+
+  useEffect(() => {
+    ref.current = value;
+  });
+
+  return ref.current;
+}
+
+export function usePixiApp(props) {
+  const { options, width, height } = props;
+  const canvasRef = useRef(null);
+  const [app, setApp] = useState(null);
```

Comment:
It doesn't look like the state is actually used anywhere.

Reply:
It's part of the return object, along with canvas.

Reply:
Right, the application is created in an effect.